Neos.Media/Classes/Domain/Repository/AssetRepository.php

```diff
@@ -334,4 +349,29 @@ public function update($object)
         parent::update($object);
         $this->assetService->emitAssetUpdated($object);
     }
+
+    /**
+     * Returns a DQL clause filtering any implementation of AssetVariantInterface
+     *
+     * @return string
+     * @var string $alias
+     */
+    protected function getAssetVariantFilterClauseForDql(string $alias): string
+    {
+        $variantClassNames = $this->reflectionService->getAllImplementationClassNamesForInterface(AssetVariantInterface::class);
+        $discriminatorTypes = array_map(
+            static function (string $variantClassName): string {
```

Comment:
would allow us to directly use the named method in the array_map here instead of declaring an inline static method that is a bit hard to decypher